src/Sentry.Unity.Editor.iOS/SentryXcodeProject.cs

```diff
@@ -0,0 +1,129 @@
+        }
+
+        public static SentryXcodeProject Open(string path)
+        {
+            return new (path);
+        }
+
+        public bool ValidateFramework()
+        {
+            if (string.IsNullOrEmpty(_relativeFrameworkPath))
+            {
+                Debug.LogWarning("Failed to locate the Sentry package within 'Frameworks'.");
+                return false;
+            }
+
+            var frameworkPath = Path.Combine(_pathToProject, _relativeFrameworkPath, FrameworkName);
+            if (!Directory.Exists(frameworkPath))
+            {
+                Debug.LogWarning($"Failed to locate 'Sentry.framework' within the Sentry package at '{frameworkPath}'.");
+                return false;
+            }
+
+            return true;
+        }
+
+        public void AddSentryFramework()
+        {
+            var targetGuid = _project.GetUnityMainTargetGuid();
+            var fileGuid = _project.AddFile(
+                Path.Combine(_relativeFrameworkPath, FrameworkName),
+                Path.Combine(_relativeFrameworkPath, FrameworkName));
+
+            var unityLinkPhaseGuid = _project.GetFrameworksBuildPhaseByTarget(targetGuid);
+
+            _project.AddFileToBuildSection(targetGuid, unityLinkPhaseGuid, fileGuid); // Link framework in 'Build Phases > Link Binary with Libraries'
+            _project.AddFileToEmbedFrameworks(targetGuid, fileGuid); // Embedding the framework because it's dynamic and needed at runtime
+
+            _project.SetBuildProperty(targetGuid, "FRAMEWORK_SEARCH_PATHS", "$(inherited)");
+            _project.AddBuildProperty(targetGuid, "FRAMEWORK_SEARCH_PATHS", $"$(PROJECT_DIR)/{_relativeFrameworkPath}/");
+
+            _project.AddBuildProperty(targetGuid, "OTHER_LDFLAGS", "-ObjC");
+        }
+
+        public void AddNativeOptions(SentryOptions options)
+        {
+            _sentryNativeOptions.CreateFile(options, Path.Combine(_pathToProject, OptionsPath));
+            _project.AddFile(OptionsPath, OptionsPath);
+        }
+
+        public void AddSentryToMain()
+        {
+            _nativeMain.AddSentry(Path.Combine(_pathToProject, MainPath));
+        }
+
+        public void Save()
+        {
+            _project.WriteToFile(_projectPath);
+        }
+
+        internal string GetRelativeFrameworkPath(string pathToProject)
+        {
+            var relativeFrameworkPath = "Frameworks/io.sentry.unity";
+            if (Directory.Exists(Path.Combine(pathToProject, relativeFrameworkPath)))
+            {
+                return Path.Combine(relativeFrameworkPath, "Plugins", "iOS");
+            }
+
+            // For dev purposes - The framework path contains the package name
+            relativeFrameworkPath += ".dev";
+            if (Directory.Exists(Path.Combine(pathToProject, relativeFrameworkPath)))
+            {
+                return Path.Combine(relativeFrameworkPath, "Plugins", "iOS");
+            }
+
+            return string.Empty;
```

Comment:
is this a valid scenario? Should we throw an exception instead or log an error and return null?